libethcore/Farm.h

```diff
@@ -83,6 +83,10 @@ class GenericFarm: public GenericFarmFace<PoW>
 	 */
 	bool start(std::string const& _sealer)
 	{
+		// call empty setWork to make sure that all workers have stoppedWorking()
+		for (auto& miner: m_miners)
+			miner->setWork();
+
```

Comment:
This thread is waiting for miner worker to stop in setWork and the miner worker can't stop because it is waiting on x_minerWork. Basically setWork must not be called with x_minerWork locked if the miner is already working, or the stopping logic changed somehow. A quick fix is make a local copy of m_miners under the lock and stop them outside of the lock. Since it is a vector of shared pointers this should be safe.